homeassistant/components/shelly/button.py

```diff
@@ -79,19 +80,61 @@ class ShellyButtonDescription(
 ]
 
 
+@callback
+def async_migrate_unique_ids(
+    entity_entry: er.RegistryEntry,
+    coordinator: ShellyRpcCoordinator | ShellyBlockCoordinator,
+) -> dict[str, Any] | None:
+    """Migrate button unique IDs."""
+    if not entity_entry.entity_id.startswith("button"):
+        return None
+
+    for key in ("reboot", "self_test", "mute", "unmute"):
+        if entity_entry.unique_id.startswith(slugify(coordinator.device.name)):
```

Comment:
It doesn't matter much here, but slugify is expense it won't change within the loop so its best to do outside the loop

Reply:
Good point, thanks